lib/llm_composer/models/open_router.ex

```diff
@@ -135,4 +134,25 @@ defmodule LlmComposer.Models.OpenRouter do
       base_request
     end
   end
+
+  defp maybe_structured_output_headers(headers, opts) do
+    has_json_schema? =
+      Keyword.has_key?(opts, :response_format) && opts.response_format.type == "json_schema"
+
+    if has_json_schema? do
+      [headers | {"Content-Type", "application/json"}]
+    else
+      headers
+    end
+  end
+
+  defp maybe_structured_output(base_request, opts) do
+    response_format = Keyword.get(opts, :response_format)
+
+    if response_format && is_map(response_format) do
```

Comment:
Can't a JSON root element be an array?

```suggestion
    if response_format && is_map(response_format) || is_list(response_format) do
```